forte/indexers/indexers.py

```diff
@@ -37,20 +40,147 @@ class BaseIndexer(ABC):
+                value are "True","wait_for", "False"
+
+            .. note::
+                "refresh" setting will greatly affect the Elasticsearch
+                performance. Please refer to
+                https://www.elastic.co/guide/en/elasticsearch/reference/master/docs-refresh.html
+                for more information on "refresh"
+        """
+        index_name = index_name if index_name else self._hparams.index_name
+        self.elasticsearch.index(  # pylint: disable=unexpected-keyword-arg
+            index=index_name, body=document, refresh=refresh)
+
+    def add_bulk(self, documents: Iterable[Dict[str, Any]],
+                 index_name: Optional[str] = None,
+                 **kwargs: Optional[Dict[str, Any]]) -> None:
+        r"""Add a bulk of documents to the index specified by ``index_name``.
+        If ``index_name`` is None, it will be picked from hparams.
+
+        Args:
+            documents (Iterable): An iterable of documents to be indexed.
+            index_name (optional, str): Name of the index where this document
+                will be saved. If None, value will be picked from hparams.
+            kwargs (optional, dict) : Optional keyword arguments like
+                "refresh", "request_timeout" etc. that are passed to
+                Elasticsearch's bulk API. Please refer to
+                https://elasticsearch-py.readthedocs.io/en/master/helpers.html#bulk-helpers
+                for the complete list of arguments.
+
+            .. note::
+                "refresh" setting will greatly affect the Elasticsearch
+                performance. Please refer to
+                https://www.elastic.co/guide/en/elasticsearch/reference/master/docs-refresh.html
+                for more information on "refresh"
+        """
+        def actions():
+            for document in documents:
+                new_document = deepcopy(document)
+                new_document.update(
+                    {"_index": index_name if index_name else
+                        self.hparams.index_name,
+                     "_type": "document"})
+                yield new_document
+
+        bulk(self.elasticsearch, actions(), **kwargs)
+
+    def search(self, query: Dict[str, Any], index_name: Optional[str] = None) \
+            -> Dict[str, Any]:
+        r"""Search the index specified by ``index_name`` that matches the
+        ``query``.
+
+        Args:
+             query (dict): An elasticseach query which is issued to the indexer
+             index_name (str): Name of the index where documents are looked up.
+                If None, value will be picked from hparams.
+
+        Returns:
+            A dict containing the documents matching the query along with
+            meta data of the search.
+        """
+        index_name = index_name if index_name else self.hparams.index_name
+        return self.elasticsearch.search(index=index_name, body=query)
+
+    def search_all(self, query: Dict[str, Any]):
```

Comment:
What's the plan for this method? If we are not planning to implement this soon, maybe we should add not_implemented error here?

Reply:
Removed it in b7632b8